cli/src/main/java/com/devonfw/tools/ide/commandlet/UpgradeSettingsCommandlet.java

```diff
@@ -0,0 +1,366 @@
+        String[] split;
+        for (String line : readLines) {
+          if (!line.contains("#") && !line.isEmpty()) {
+            if (line.contains("DEVON_IDE_CUSTOM_TOOLS")) {
+              createCustomToolsJson(line);
+              continue;
+            }
+            if (line.contains("DEVON_")) {
+              line = line.replace("DEVON_", "");
+            }
+            split = line.split("[ =]");
+            if (split.length == 3) {
+              devonProperties.put(split[1], new String[] { split[0], split[2] });
+            }
+            if (split.length == 2) {
+              devonProperties.put(split[0], split[1]);
+            }
+          }
+        }
+
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+
+      if (context.getFileAccess().findFirst(file_path.getParent(), path -> path.getFileName().toString().equals("ide.properties"), false) != null) {
+        try {
+          List<String> readLines = Files.readAllLines(target);
+          String[] split;
+
+          Properties ideProperties = new Properties();
+          for (String line : readLines) {
+            if (!line.contains("#") && !line.isEmpty()) {
+              split = line.split("[ =]");
+              if (split.length == 3) {
+                ideProperties.put(split[1], new String[] { split[0], split[2] });
+              }
+              if (split.length == 2) {
+                ideProperties.put(split[0], split[1]);
+              }
+            }
+          }
+
+          Properties mergedProperties = new Properties();
+          mergedProperties.putAll(ideProperties.entrySet().stream().collect(Collectors.toMap(e -> e.getKey().toString(), e -> e.getValue())));
+          mergedProperties.putAll(devonProperties.entrySet().stream().collect(Collectors.toMap(e -> e.getKey().toString(), e -> e.getValue())));
+          for (Entry<Object, Object> set : ideProperties.entrySet()) {
+            mergedProperties.remove(set.getKey());
+          }
+
+          for (Entry<Object, Object> set : mergedProperties.entrySet()) {
+            if (set.getValue() instanceof String) {
+              Files.write(target, ("\n" + set.getKey().toString() + "=" + set.getValue().toString()).getBytes(), StandardOpenOption.APPEND);
+            }
+            if (set.getValue() instanceof String[]) {
+
+              String[] values = (String[]) set.getValue();
+              Files.write(target, ("\n" + values[0] + " " + set.getKey().toString() + "=" + values[1]).getBytes(), StandardOpenOption.APPEND);
+            }
+          }
+
+          this.context.success("Successfully merged and updated ide.properties: " + file_path);
+
+          Files.delete(file_path);
+
+        } catch (IOException e) {
+          throw new RuntimeException(e);
+        }
+      } else {
+        try {
+          String comment = "#********************************************************************************\n"
```

Comment:
If you need something like this, you should use [multi-line-strings](https://www.baeldung.com/java-multiline-string).